proparse/src/main/java/org/prorefactor/treeparser01/TP01Support.java

```diff
@@ -516,17 +528,16 @@ public void widattr(AST widAST, AST idAST, ContextQualifier cq) throws TreeParse
           result.variable.noteReference(cq);
         }
       }
-    } else if (idAST.getType() == NodeTypes.Field_ref) {
+    } else if (idNode.getType() == ProParserTokenTypes.Field_ref) {
       // Reference to a static field
-      JPNode idNode = (JPNode) idAST;
-      if ((idNode.getFirstChild().getType()) == NodeTypes.ID && (idNode.nextSibling() != null) && (idNode.nextSibling().getType() == NodeTypes.OBJCOLON)) {
-        String clsRef = idAST.getFirstChild().getText();
+      if ((idNode.getFirstChild().getType()) == ProParserTokenTypes.ID && (idNode.getNextSibling() != null) && (idNode.getNextSibling().getType() == ProParserTokenTypes.OBJCOLON)) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1066)